pact-jvm-provider-sbt/src/main/scala/au/com/dius/pact/provider/sbt/SbtProviderPlugin.scala

```diff
@@ -12,27 +12,30 @@ import sbt._
 import scala.collection.JavaConversions
 import scala.collection.mutable.ArrayBuffer
 
-object SbtProviderPlugin extends Plugin {
-
-  val providers = SettingKey[Seq[ProviderConfig]]("providers", "Providers to verify")
-  val pactVerify = taskKey[Unit]("Verify the pacts for all defined providers")
-
-  val config = Seq(
-    providers := Seq(),
-    pactVerify := {
-      val s: TaskStreams = Keys.streams.value
-
-      if (System.getProperty("pact.logLevel") != null) {
-        LoggerFactory.getLogger(org.slf4j.Logger.ROOT_LOGGER_NAME).asInstanceOf[Logger].setLevel(
-          Level.toLevel(System.getProperty("pact.logLevel"))
-        )
-      }
-
-      if (providers.value.isEmpty)
-        sys.error("No providers have been defined. Configure them by setting the providers build setting.")
-      else
-        Verification.verify(providers.value)
-    })
+object SbtProviderPlugin extends AutoPlugin {
+  object autoImport {
+    lazy val pactProviders = SettingKey[Seq[ProviderConfig]]("providers", "Providers to verify")
```

Comment:
note: `providers` is renamed to `pactProviders`, and `config` to `pactProvidersConfig`. The names are only suggestions, but since the AutoPlugin brings all of that in scope automatically, names need to be more specific. Unfortunately this breaks the old API.